applications/OptimizationApplication/python_scripts/filtering/filtering_factory.py

```diff
@@ -0,0 +1,35 @@
+import typing
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import FilterTypes
+
+def Factory(model: Kratos.Model, filtering_model_part_name: str, variable: SupportedSensitivityFieldVariableTypes, data_location: Kratos.Globals.DataLocation, settings: Kratos.Parameters) -> FilterTypes:
+    if not settings.Has("filter_type"):
+        raise RuntimeError(f"\"filter_type\" not specified in settings:\n{settings}")
+
+    if not settings.Has("filtering_model_part_name"):
+        settings.AddString("filtering_model_part_name", filtering_model_part_name)
+
+    if settings["filtering_model_part_name"].GetString() != filtering_model_part_name:
```

Comment:
why do you need to give filtering_model_part_name two times, in settings and as variable?

Reply:
this is removed. sorry for not informing, this branch was a testing branch. I created the fresh branch in #12142